src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcShopDao.java

```diff
@@ -3,83 +3,43 @@
 import com.vinylteam.vinyl.dao.ShopDao;
 import com.vinylteam.vinyl.dao.jdbc.mapper.ShopRowMapper;
 import com.vinylteam.vinyl.entity.Shop;
-import com.zaxxer.hikari.HikariDataSource;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
 
-import java.sql.Connection;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
 import java.util.ArrayList;
 import java.util.List;
 
 @Slf4j
+@Repository
+@RequiredArgsConstructor
 public class JdbcShopDao implements ShopDao {
 
-    private static final ShopRowMapper shopRowMapper = new ShopRowMapper();
     private static final String SELECT_SHOPS_BY_IDS = "SELECT id, link_to_main_page, link_to_image, name, link_to_small_image " +
-            "FROM shops WHERE id IN ()  ORDER BY shop_order NULLS FIRST";
+            "FROM shops WHERE id IN (:ids) ORDER BY shop_order NULLS FIRST";
 
     private static final String SELECT_ALL_SHOPS = "SELECT id, link_to_main_page, link_to_image, name, link_to_small_image FROM shops ORDER BY shop_order NULLS FIRST";
 
-    private final HikariDataSource dataSource;
-
-    public JdbcShopDao(HikariDataSource dataSource) {
-        this.dataSource = dataSource;
-    }
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+    private final RowMapper<Shop> rowMapper = new ShopRowMapper();
 
     @Override
-    public List<Shop> getManyByListOfIds(List<Integer> ids) {
-        List<Shop> shops = new ArrayList<>();
-        if (!ids.isEmpty()) {
-            String queryToExecute = fillSelectManyByIdsStatement(ids);
-            try (Connection connection = dataSource.getConnection();
-                 Statement getManyByListOfIdsStatement = connection.createStatement();
-                 ResultSet resultSet = getManyByListOfIdsStatement.executeQuery(queryToExecute)) {
-                log.debug("Executed statement {'statement':{}}", getManyByListOfIdsStatement);
-                while (resultSet.next()) {
-                    Shop shop = shopRowMapper.mapRow(resultSet);
-                    shops.add(shop);
-                }
-            } catch (SQLException e) {
-                log.error("Error while getting list of shops by ids list from db {'ids':{}, 'shops':{}}",
-                        ids, shops, e);
-                throw new RuntimeException(e);
-            }
+    public List<Shop> findByListOfIds(List<Integer> ids) {
+        if (ids.isEmpty()) {
+            return new ArrayList<>();
```

Comment:
Maybe you should check  ids in servise and remove this part?

Reply:
Unfortunately, sql request WHERE id IN() crashes without values in (), so I'll leave it like that for now.